examples/provisioned.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+
+use Illuminate\Database\DatabaseManager;
+use SaatchiArt\BentoBoxDDD\Adapters\Repositories\ArtworkRepositoryMysqlImpl;
+use SaatchiArt\BentoBoxDDD\Adapters\Repositories\Transactions\TransactionFactory;
+use SaatchiArt\BentoBoxDDD\Adapters\Repositories\UserRepositoryMysqlImpl;
+use SaatchiArt\BentoBoxDDD\Domain\Repositories\ArtworkRepository;
+use SaatchiArt\BentoBoxDDD\Domain\Repositories\UserRepository;
+
+class SomeService {
+    private ArtworkRepository $artworkRepository;
+    private UserRepository $userRepository;
+    private TransactionFactory $transactionFactory;
+
+    public function __construct(
+        ArtworkRepository $artworkRepository,
+        UserRepository $userRepository,
+        TransactionFactory $transactionFactory
+    ) {
+        $this->artworkRepository = $artworkRepository;
+        $this->userRepository = $userRepository;
+        $this->transactionFactory = $transactionFactory;
+    }
+
+    public function updateThoseArtworksForSomeReason(): int
+    {
+        $transaction = $this->transactionFactory->makeFullyProvisionedTransaction(
+            [$this->artworkRepository, $this->userRepository],
```

Comment:
By provisioning it with the repositories (or transaction providers), the provisioner can check/verify they share the same connection.